File: rust/src/models/list_response_endpoint_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::endpoint_out::EndpointOut;

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct ListResponseEndpointOut {
    pub data: Vec<EndpointOut>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub iterator: Option<String>,

    #[serde(rename = "prevIterator")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub prev_iterator: Option<String>,

    pub done: bool,
}

impl ListResponseEndpointOut {
    pub fn new(data: Vec<EndpointOut>, done: bool) -> Self {
        Self {
            data,
            iterator: None,
            prev_iterator: None,
            done,
        }
    }
}

File: javascript/src/models/listResponseEndpointOut.ts
// this file is @generated
import { type EndpointOut, EndpointOutSerializer } from "./endpointOut";

export interface ListResponseEndpointOut {
  data: EndpointOut[];
  iterator: string | null;
  prevIterator?: string | null;
  done: boolean;
}

export const ListResponseEndpointOutSerializer = {
  _fromJsonObject(object: any): ListResponseEndpointOut {
    return {
      data: object["data"].map((item: EndpointOut) =>
        EndpointOutSerializer._fromJsonObject(item)
      ),
      iterator: object["iterator"],
      prevIterator: object["prevIterator"],
      done: object["done"],
    };
  },

  _toJsonObject(self: ListResponseEndpointOut): any {
    return {
      data: self.data.map((item) => EndpointOutSerializer._toJsonObject(item)),
      iterator: self.iterator,
      prevIterator: self.prevIterator,
      done: self.done,
    };
  },
};
